snuba/web/split.py

```diff
@@ -128,61 +158,87 @@ def time_split(dataset, request: Request, *args, **kwargs):
 
         return overall_result
 
-    def col_split(
-        dataset, request: Request, column_split_spec: ColumnSplitSpec, *args, **kwargs
-    ):
+
+class ColumnSplitQueryStrategy(StorageQuerySplitStrategy):
+    """
+    A strategy that performs column based splitting: if the client requests enough columns,
+    a first query on the minimum set of columns is ran to load as little Clickhouse data
+    as possible. A second query based on the results of the first is then executed to
+    build the full result set.
+    """
+
+    def __init__(
+        self, id_column: str, project_column: str, timestamp_column: str
+    ) -> None:
+        self.__id_column = id_column
+        self.__project_column = project_column
+        self.__timestamp_column = timestamp_column
+
+    def __get_min_columns(self) -> Sequence[str]:
+        return [self.__id_column, self.__project_column, self.__timestamp_column]
```

Comment:
This is only used once now, I'm not sure it's beneficial to make it its own method.